internal/services/tesla_fleet_api_service.go

```diff
@@ -247,6 +269,56 @@ func (t *teslaFleetAPIService) RefreshToken(ctx context.Context, refreshToken st
 	return tokResp, nil
 }
 
+var fields = TelemetryFields{
+	"ChargeState":         {IntervalSeconds: 300},
+	"Location":            {IntervalSeconds: 10},
+	"OriginLocation":      {IntervalSeconds: 300},
+	"DestinationLocation": {IntervalSeconds: 300},
+	"DestinationName":     {IntervalSeconds: 300},
+	"EnergyRemaining":     {IntervalSeconds: 300},
+	"VehicleSpeed":        {IntervalSeconds: 60},
+	"Odometer":            {IntervalSeconds: 300},
+	"EstBatteryRange":     {IntervalSeconds: 300},
+	"Soc":                 {IntervalSeconds: 300},
+	"BatteryLevel":        {IntervalSeconds: 60},
+}
+
+func (t *teslaFleetAPIService) SubscribeForTelemetryData(ctx context.Context, token, region, vin string) error {
+	baseURL := fmt.Sprintf(t.Settings.TeslaFleetURL, region)
+	u, err := url.JoinPath(baseURL, "/api/1/vehicles/fleet_telemetry_config")
+	if err != nil {
+		return err
+	}
+
+	exp := time.Now().AddDate(0, 0, 364).Unix()
+
+	r := SubscribeForTelemetryDataRequest{
+		Vins: []string{vin},
+		Config: TelemetryConfigRequest{
+			HostName:            t.Settings.TeslaTelemetryHostName,
+			PublicCACertificate: t.Settings.TeslaTelemetryCACertificate,
+			Expiration:          exp,
+			Port:                t.Settings.TeslaTelemetryPort,
+			Fields:              fields,
+			AlertTypes:          []string{"service"},
+		},
+	}
+
+	b, err := json.Marshal(r)
+	if err != nil {
+		return err
+	}
+
+	resp, err := t.performRequest(ctx, u, token, http.MethodPost, bytes.NewReader(b))
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	return nil
```

Comment:
Mustn't we check the response? See their sample:
```
{
 "response": {
  "updated_vehicles": 1,
  "skipped_vehicles": {
   "missing_key": [],
   "unsupported_hardware": [
    "5YJ3000000NEXUS02"
   ],
   "unsupported_firmware": [
    "5YJ3000000NEXUS02"
   ]
  }
 }
}
```

Reply:
I guess in this case we can look for `updated_vehicles` being 1, and if not we can see what list it's on.